enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,530 @@
+In MicroShift, however, `Managed`` does not have the exact same meaning, as
+the management of the resources happens only when MicroShift starts. Any
+changes made to the router not using MicroShift's configuration will get
+reverted on the next restart.
+
+### Listening ports
+The following configuration is proposed:
+```yaml
+ingress:
+  ports:
+    http: <uint16> # Defaults to 80.
+    https: <uint16> # Defaults to 443.
+```
+
+MicroShift does not own the host, which means there might be other
+applications running alongside and some ports may already be opened.
+MicroShift must be able to accommodate such situations and be ready to listen
+in other ports, hence the configuration options.
+
+In order to allow this configuration and many other advantages, the
+router service shall be changed so that it is exposed using `LoadBalancer`
+type instead of using host ports. See the [Design Details section](#why-loadbalancer-service)
+for more information.
+
+Possible values are restricted to 1-65535.
+
+#### Firewalling ports
+Using `LoadBalancer` service type prevents the usage of firewalld to block
+access to ports. Since this is a feature that users may require for auditing,
+an alternative is provided.
+
+The `NetworkPolicy` resource fits the needs for blocking traffic to services
+and pods. Policies can also be configured to allow specific IP addresses and
+provide a high degree of customization for network traffic. They use the CNI
+instead of host level settings, providing greater flexibility and being self
+contained within MicroShift.
+
+Users in need of blocking traffic shall use the `NetworkPolicy` resource. A
+section in the documentation will be added to describe how to do this.
+
+### Configure IPs where the router listens
+The following configuration is proposed:
+```yaml
+ingress:
+  expose: # Defaults to IPs in the host. Details below.
+    - <NIC name|IP address>
+    - ...
+```
+
+As described in [this section](#using-loadbalancer-service-type), the use of
+`LoadBalancer` makes ovnk configure iptables rules to expose the service
+outside of the cluster.
+Building on [LoadBalancer controller enhancement](loadbalancer-service-support.md),
+the service is exposed using the node IP and configured ports.
+
+All the IPs where the router is exposed need to be part of the `LoadBalancer`
+service status definition. In order to simplify configuration the router will
+allow specifying only IP addresses and interface names.
+
+MicroShift will default to listen to external IPs, the service IP and the
+apiserver IP, as this was the previous behavior and it needs to keep it for
+compatibility. Any advanced configuration, such as multiple interfaces, VLANs,
+etc. will need user's configuration. Any user's configuration will override all
+of the defaults.
+
+If the configuration includes duplicates in the form of same entries in the
+same list or referring to the same IP in different ways (hostnames or NICs),
```

Comment:
Here and a couple other places, you still mention hostnames even though you removed the option to specify hostnames in a recent change.

Reply:
Done